### PR TITLE
[Feature][ZETA REST API] Can submit job with rest api to Zeta master node automaticly

### LINKED ISSUE #5862: [Feature][ZETA REST API] CAN submit job with rest api to Zeta master node automaticly
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

now ,when i submit job with rest api to Zeta ,it return can not submit requst to inactive master node, CAN submit job with rest api to Zeta master node automaticly,Users do not need to worry about whether it is a master node

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liugddx**
### Purpose of this pull request

close #5862

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — COMMENTED** [review]
Overall LGTM. But the variable names are too arbitrary, such as `hazelcastInstance1`, `hazelcastInstance2`, `seaTunnelServer1`. Please do some change.

**Comment by liugddx** [comment]
> Overall LGTM. But the variable names are too arbitrary, such as `hazelcastInstance1`, `hazelcastInstance2`, `seaTunnelServer1`. Please do some change.

ok i modified it.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 1]
```diff
@@ -83,242 +92,323 @@ void beforeClass() throws Exception {
...
-                                + RestConstant.RUNNING_JOB_URL
-                                + "/"
-                                + clientJobProxy.getJobId())
-                .then()
-                .statusCode(200)
-                .body("jobName", equalTo("fake_to_file"))
-                .body("jobStatus", equalTo("RUNNING"));
+        Arrays.asList(node2, node1)
+                .forEach(
+                        hazelcastInstance2 -> {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]:
```suggestion
                        instance -> {
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 2]
```diff
@@ -83,242 +92,323 @@ void beforeClass() throws Exception {
...
-                                        .getAddress()
-                                        .getPort()
-                                + RestConstant.RUNNING_JOBS_URL)
-                .then()
-                .statusCode(200)
-                .body("[0].jobName", equalTo("fake_to_file"))
-                .body("[0].jobStatus", equalTo("RUNNING"));
+        Arrays.asList(node2, node1)
+                .forEach(
+                        hazelcastInstance2 -> {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]: ditto

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 3]
```diff
@@ -83,242 +92,323 @@ void beforeClass() throws Exception {
...
-                                        .getAddress()
-                                        .getPort()
-                                + RestConstant.SYSTEM_MONITORING_INFORMATION)
-                .then()
-                .assertThat()
-                .time(lessThan(5000L))
-                .statusCode(200);
+        Arrays.asList(node2, node1)
+                .forEach(
+                        hazelcastInstance2 -> {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]: ditto

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 4]
```diff
@@ -83,242 +92,323 @@ void beforeClass() throws Exception {
...
-                .untilAsserted(
-                        () ->
-                                Assertions.assertEquals(
-                                        JobStatus.FINISHED,
-                                        seaTunnelServer
-                                                .getCoordinatorService()
-                                                .getJobStatus(Long.parseLong(jobId))));
+        Arrays.asList(node2, node1)
+                .forEach(
+                        hazelcastInstance2 -> {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]: ditto

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 5]
```diff
@@ -83,242 +92,323 @@ void beforeClass() throws Exception {
...
+                            Response response = submitJob(hazelcastInstance2, "BATCH");
+                            response.then().statusCode(200).body("jobName", equalTo("test测试"));
+                            String jobId = response.getBody().jsonPath().getString("jobId");
+                            SeaTunnelServer seaTunnelServer = null;
+
+                            for (HazelcastInstance hazelcastInstance :
+                                    Hazelcast.getAllHazelcastInstances()) {
+                                SeaTunnelServer seaTunnelServer1 =
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]:
```suggestion
                                SeaTunnelServer server =
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 6]
```diff
@@ -83,242 +92,323 @@ void beforeClass() throws Exception {
...
+                                    submitJob(hazelcastInstance2, "STREAMING")
+                                            .getBody()
+                                            .jsonPath()
+                                            .getString("jobId");
+                            SeaTunnelServer seaTunnelServer = null;
 
-        String jobId2 = submitJob("STREAMING").getBody().jsonPath().getString("jobId");
+                            for (HazelcastInstance hazelcastInstance :
+                                    Hazelcast.getAllHazelcastInstances()) {
+                                SeaTunnelServer seaTunnelServer1 =
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]: ditto

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java [hunk 7]
```diff
@@ -83,242 +92,323 @@ void beforeClass() throws Exception {
...
-                .untilAsserted(
-                        () ->
-                                Assertions.assertEquals(
-                                        JobStatus.FINISHED,
-                                        seaTunnelServer
-                                                .getCoordinatorService()
-                                                .getJobStatus(Long.parseLong(jobId))));
+        Arrays.asList(node2, node1)
+                .forEach(
+                        hazelcastInstance2 -> {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/RestApiIT.java]: ditto

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpPostCommandProcessor.java [hunk 8]
```diff
@@ -92,7 +95,25 @@ public void handle(HttpPostCommand httpPostCommand) {
     private SeaTunnelServer getSeaTunnelServer() {
         Map<String, Object> extensionServices =
                 this.textCommandService.getNode().getNodeExtension().createExtensionServices();
-        return (SeaTunnelServer) extensionServices.get(Constant.SEATUNNEL_SERVICE_NAME);
+        SeaTunnelServer seaTunnelServer =
+                (SeaTunnelServer) extensionServices.get(Constant.SEATUNNEL_SERVICE_NAME);
+        if (!seaTunnelServer.isMasterNode()) {
+            for (HazelcastInstance hazelcastInstance : Hazelcast.getAllHazelcastInstances()) {
+                SeaTunnelServer seaTunnelServer1 =
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpPostCommandProcessor.java]: ditto

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java [hunk 9]
```diff
@@ -48,5 +48,7 @@ public class RestConstant {
     public static final String SYSTEM_MONITORING_INFORMATION =
             "/hazelcast/rest/maps/system-monitoring-information";
 
+    public static final String HTTP_SUBMIT_JOB = "http://%s:%s";
```
Comment (zhilinli123) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java]: No call?

**Review by Hisoka-X — APPROVED** [review]
LGTM if test passed

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**